exonum/src/storage/proof_map_index/key.rs

```diff
@@ -170,17 +171,20 @@ impl DBKey {
         if self.from != other.from {
             0
         } else {
+            let mut max_len = min(self.len(), other.len());
+
             let from = self.from / 8;
             let to = min((self.to + 7) / 8, (other.to + 7) / 8);
-            let max_len = min(self.len(), other.len());
 
             for i in from..to {
                 let x = self.data[i as usize] ^ other.data[i as usize];
                 if x != 0 {
-                    let tail = x.leading_zeros() as u16;
-                    return min(i * 8 + tail - self.from, max_len);
+                    let tail = x.trailing_zeros() as u16;
+                    max_len = min(i * 8 + tail - self.from, max_len);
```

Comment:
I think `return min()` would read slightly better in this case, and will allow to reduce the difference in this function to just `s/leading/trailing/`, because you won't need `mut` anymore.